perfact/zodbsync/zodbsync.py

```diff
@@ -704,14 +690,18 @@ def playback_paths(self, paths, recurse=True, override=False,
         txn_mgr = self.start_transaction(note=note)
 
         try:
+            results = []
             for path in paths:
-                self.playback(
+                results.append(self.playback(
                     path=path,
                     override=override,
                     recurse=recurse,
                     skip_errors=skip_errors,
                     encoding=encoding,
-                )
+                ))
+            for result in reversed(results):
+                if result:
+                    self.playback_after_recurse(**result)
```

Comment:
Oh, so doing the hook in reverse order is what fixes the Ordered Folder playback issue?

Reply:
Correct, see my comment in #83 - the ordering of an ordered folder was adjusted before its (new) children were even present, if they were created in the same operation, so they were not respected.

Not sure if this fixes all issues with Ordered Folders, but this was the issue that I could reproduce.

The refactoring I would like to see I started in #86, but did not finish yet. I will continue that when I get to it, but this bugfix should come first.

Reply:
Slight correction: it is not that the hook is executed in reverse order. It is that the hook for the parent is executed only after all its children are created. So the `reversed` could it principle also be removed, as long the hook is executed in a separate loop.

But in the recursion case the hook was executed on the "way up", so to keep it consistent I included the `reversed`.